src/main/java/taxi/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package taxi.service;
+
+import java.util.Optional;
+import taxi.exception.AuthenticationException;
+import taxi.lib.Inject;
+import taxi.lib.Service;
+import taxi.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverService driverService;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Optional<Driver> driver = driverService.findByLogin(login);
+        if (driver.isEmpty()) {
+            throw new AuthenticationException("Login or password was incorrect.");
+        }
+        if (driver.get().getPassword().equals(password)) {
+            return driver.get();
+        }
```

Comment:
Both in AuthenticationServiceImpl and in AuthenticationFilter, use only one if block if possible.
[checklist](https://mate-academy.github.io/jv-program-common-mistakes/java-web/web-security/java-web-security)